### PR TITLE
Candidate fix for #11811

### LINKED ISSUE #11811: Feature: databinding java.time.Instant from String
Currently, Grails can only databind java.time.Instant from number (e.g. 1620809710000 - milliseconds since Epoch), but I prefer to send java.time.Instant in a human readable format (e.g. "2021-05-12T08:13:02.907871Z") and java.time.Instant.parse(String) can be used for that.

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Create a new Grails 4 app
2. Add a controller with a command object that has a field of type java.time.Instant
3. Start the app and call the controller with JSON payload that has java.time.Instant string - e.g. "2021-05-12T08:13:02.907871Z"
See also README.md for https://github.com/andersaaberg/grails-bind-instant-string

### Expected Behaviour

The java.time.Instant field should be data-binded

### Actual Behaviour

The java.time.Instant field is null

### Environment Information

- **Operating System**: Ubuntu 20.04
- **Grails Version:** 4.0.10
- **JDK Version:** openjdk version "1.8.0_275"

### Example Application

https://github.com/andersaaberg/grails-bind-instant-string

I have already fixed this and will submit a PR ASAP.

### PR DESCRIPTION
**Pull request by andersaaberg**
Enable databinding of java.time.Instant from string and not just from number

### CODE REVIEW

File: grails-plugin-databinding/src/main/groovy/org/grails/databinding/converters/Jsr310ConvertersConfiguration.groovy [hunk 1]
```diff
@@ -321,6 +321,25 @@ class Jsr310ConvertersConfiguration {
         }
     }
 
+    @Bean
+    ValueConverter instantStringValueConverter() {
+        new ValueConverter() {
+            @Override
+            boolean canConvert(Object value) {
```
Comment (sbglasius) [inline grails-plugin-databinding/src/main/groovy/org/grails/databinding/converters/Jsr310ConvertersConfiguration.groovy:328]: If `convert` casts to `CharSequence` then `canConvert` should check if `value` is instanceof `CharSequence`

Comment (andersaaberg) [reply grails-plugin-databinding/src/main/groovy/org/grails/databinding/converters/Jsr310ConvertersConfiguration.groovy:328]: Good point, "String" is an unnecessary restriction

**Comment by andersaaberg** [comment]
> @andersaaberg Do you know why the tests are failing?

@puneetbehl  My apologies, the failing test did not seem to be associated with my changes so I just ignored it. I have rebased from master and run "./gradlew check" without getting any errors. I have pushed this and hopefully the tests on github will pass as well.

**Comment by andersaaberg** [comment]
> > @andersaaberg Do you know why the tests are failing?
> 
> @puneetbehl My apologies, the failing test did not seem to be associated with my changes so I just ignored it. I have rebased from master and run "./gradlew check" without getting any errors. I have pushed this and hopefully the tests on github will pass as well.

Now no tests are failing - the build is only failing due to permissions "Error: Resource not accessible by integration" which is probably because I do not have permissions or the build is from a fork.

**Pull request merged**